### PR TITLE
pam-cgfs: ignore the system umask when creating the cgroup hierarchy

### PR DESCRIPTION
**Pull request by 3XX0**
Fixes: #2277
Signed-off-by: Jonathan Calmels <jcalmels@nvidia.com>

### CODE REVIEW

**Comment by hallyn** [comment]
jenkins: test this please

File: src/lxc/pam/pam_cgfs.c [hunk 1]
```diff
@@ -223,6 +223,17 @@ static bool cgv2_prune_empty_cgroups(const char *user);
 static bool cgv2_remove(const char *cgroup);
 static bool is_cgv2(char *line);
 
+static int do_mkdir(const char *path, mode_t mode)
+{
+        mode_t mask;
+        int r;
+
+        mask = umask(0);
+        r = mkdir(path, mode);
```
Comment (hallyn) [inline src/lxc/pam/pam_cgfs.c]: Looks fine - though how about keeping and resetting saved_errno?

Comment (3XX0) [reply src/lxc/pam/pam_cgfs.c]: Yeah might as well, although `umask` doesn't touch it, does it?

Comment (brauner) [reply src/lxc/pam/pam_cgfs.c]: Well, POSIX doesn't give a guarantee ;)

Comment (3XX0) [reply src/lxc/pam/pam_cgfs.c]: Fair enough, I pushed the change + rebased on master

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
Thanks @3XX0 :)

**Pull request merged**